rosdoc2/verbs/build/builders/doxygen_builder.py

```diff
@@ -149,6 +157,13 @@ def __init__(self, builder_name, builder_entry_dictionary, build_context):
             logger.info(f"Using user specified Doxyfile at '{self.doxyfile}'.")
 
     def build(self, *, doc_build_folder, output_staging_directory):
+        # If the build type is `ament_python`, there is no reason to run doxygen.
+        if self.build_context.build_type == 'ament_python':
```

Comment:
Sorry, I hadn't understood the `mixed` option yet. So maybe this comment doesn't apply.

Reply:
If we drop the `mixed` option and stick to the `ament_cmake`/`ament_python` split for now, would it be reasonable to change this condition to

```suggestion
        if 'cmake' not in self.build_context.build_type:  # ['cmake', 'ament_cmake']
```
?
I'm still not entirely sure about the `cmake` build type altogether, so maybe instead

```suggestion
        if self.build_context.build_type != 'ament_cmake'
```
if the pure `cmake` build type is not supported?

Reply:
For the cmake build type and any other build type that we don't explicitly support here I'd say we just try to run Sphinx if there is a doc folder with a conf.py. Otherwise do nothing special. We can revisit that later, improve the story for cmake and other build types later.

Reply:
In 195c286, I've bundled `cmake` build type's behavior with that of the `ament_cmake` build type, i.e. they behave identically. I believe that's a reasonable assumption for such build types.

Reply:
As it stands, the `ament_cmake` and `cmake` build types behave identically in `rosdoc2`. One example:

```py
# If the build type is not `ament_cmake/cmake`, there is no reason
# to create a doxygen builder.
if (self.build_context.build_type not in ('ament_cmake', 'cmake') and
```

However, the structure as-is allows different behaviors easily. Should I resolve this conversation?

Reply:
The only important difference is that with `ament_cmake` we can more reasonably assume a particular layout of the files in the project. You find much more variety in plain cmake packages.

That's not really an issue I guess though since we can't fully assume anything in either case. So we can lump `cmake` in with `ament_cmake`.